MCprep_addon/spawner/effects.py

```diff
@@ -0,0 +1,1180 @@
+		and blend.lower().endswith("blend")
+	]
+
+	for bfile in blends:
+		with bpy.data.libraries.load(bfile) as (data_from, _):
+			collections = spawn_util.filter_collections(data_from)
+
+		for itm in collections:
+			effect = mcprep_props.effects_list.add()
+			effect.effect_type = "collection"
+			effect.name = itm
+			effect.filepath = bfile
+			effect.index = len(mcprep_props.effects_list) - 1  # For icon index.
+
+
+def load_image_sequence_effects(context):
+	"""Load effects from the particles folder that should be animated."""
+	mcprep_props = context.scene.mcprep_props
+
+	resource_folder = bpy.path.abspath(context.scene.mcprep_texturepack_path)
+
+	# Check levels
+	lvl_0 = os.path.join(resource_folder, "particle")
+	lvl_1 = os.path.join(resource_folder, "textures", "particle")
+	lvl_2 = os.path.join(resource_folder, "minecraft", "textures", "particle")
+	lvl_3 = os.path.join(resource_folder, "assets", "minecraft", "textures", "particle")
+
+	if not os.path.isdir(resource_folder):
+		conf.log("Error, resource folder does not exist")
+		return
+	elif os.path.isdir(lvl_0):
+		resource_folder = lvl_0
+	elif os.path.isdir(lvl_1):
+		resource_folder = lvl_1
+	elif os.path.isdir(lvl_2):
+		resource_folder = lvl_2
+	elif os.path.isdir(lvl_3):
+		resource_folder = lvl_3
+
+	# List all files and reduce to basename: fname_01.png to fname.
+	rfiles = os.listdir(resource_folder)
+	effect_files = [
+		os.path.splitext(fname)[0].rsplit("_", 1)[0]
+		for fname in rfiles
+		if os.path.isfile(os.path.join(resource_folder, fname))
+		and os.path.splitext(fname)[-1].lower() in EXTENSIONS
+	]
+	effect_files = list(set(effect_files))
+
+	effect_files = sorted(effect_files)
+
+	for itm in effect_files:
+		effect = mcprep_props.effects_list.add()
+		effect.effect_type = "img_seq"
+		effect.name = itm.replace("_", " ").capitalize()
+		effect.description = "Instance the {} particle aniamted sequence".format(
+			itm)
+
+		# Note: the item being added is NOT a full filepath, but the prefix.
+		effect.filepath = os.path.join(resource_folder, itm)
+
+		effect.index = len(mcprep_props.effects_list) - 1  # For icon index.
+
+		# Try to load a middle frame as the icon.
+		if not conf.use_icons or conf.preview_collections["effects"] == "":
+			continue
+		effect_files = [
+			os.path.join(resource_folder, fname)
+			for fname in rfiles
+			if os.path.isfile(os.path.join(resource_folder, fname))
+			and fname.startswith(itm)
+			and os.path.splitext(fname)[-1].lower() in EXTENSIONS
+		]
+		if effect_files:
+			# 0 if 1 item, otherwise greater side of median.
+			e_index = int(len(effect_files) / 2)
+			conf.preview_collections["effects"].load(
+				"effects-{}".format(effect.index), effect_files[e_index], 'IMAGE')
+
+
+# -----------------------------------------------------------------------------
+# Support functions
+# -----------------------------------------------------------------------------
+
+
+def create_auto_footfall_particle_plane():
```

Comment:
Seems like this doesn't do much. It may be best to comment this out so Python doesn't have to create a function object

Reply:
Fair call - better to make it an enhancement issue vs deadcode, removed in the meantime.